lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelValidation.cxx

```diff
@@ -0,0 +1,170 @@
+{
+  if ((cvMethod != LEAVEONEOUT) and (cvMethod != KFOLD))
+    throw InvalidArgumentException(HERE) << "The method " << cvMethod << " is not available.";
+  cvMethod_ = cvMethod;
+  if (kParameter < 1)
+    throw InvalidArgumentException(HERE) << "Cannot set k parameter of K-Fold method to " << kParameter << " which is lower than 1";
+  const UnsignedInteger sampleSize = linearModelResult_.getSampleResiduals().getSize();
+  if (kParameter > sampleSize)
+    throw InvalidArgumentException(HERE) << "Cannot set k parameter of K-Fold method to " << kParameter 
+      << " which is larger than the sample size =" << sampleSize;
+  kParameter_ = kParameter;
+}
+
+/* Virtual constructor */
+LinearModelValidation * LinearModelValidation::clone() const
+{
+  return new LinearModelValidation(*this);
+}
+
+
+/* String converter */
+String LinearModelValidation::__repr__() const
+{
+  return OSS(true)
+         << "class=" << getClassName()
+         << ", linearModelResult=" << linearModelResult_
+         << ", kParameter_=" << kParameter_
+         << ", cvMethod_=" << cvMethod_;
+}
+
+/* Compute cross-validation predictions */
+Sample LinearModelValidation::ComputeMetamodelCrossValidationPredictions(
+    const LinearModelResult & linearModelResult, const CrossValidationMethod cvMethod, 
+    const UnsignedInteger & kParameter)
+{
+  // The residuals is ri = g(xi) - tilde{g}(xi) where g is the model
+  // and tilde(g) is the metamodel.
+  // Hence the metamodel prediction is tilde{g}(xi) = yi - ri.
+  const Sample residualsSample(linearModelResult.getSampleResiduals());
+  const UnsignedInteger sampleSize = residualsSample.getSize();
+  const UnsignedInteger basisSize = linearModelResult.getBasis().getSize();
+  if (basisSize > sampleSize)
+    throw InvalidArgumentException(HERE) << "Error: the sample size is: " << sampleSize << 
+      " which is lower than the basis size: " << basisSize;
+  const Sample outputSample(linearModelResult.getOutputSample());
+  Sample cvPredictions(sampleSize, 1);
+  if (cvMethod == LEAVEONEOUT)
+  {
+    const Point diagonalH(linearModelResult.getLeverages());
+    for (UnsignedInteger i = 0; i < sampleSize; ++i)
+      cvPredictions(i, 0) = outputSample(i, 0) - residualsSample(i, 0) / (1.0 - diagonalH[i]);
+  }
+  else if (cvMethod == KFOLD)
+  {
+    SymmetricMatrix projectionMatrix(linearModelResult.computeProjectionMatrix());
+    KFoldSplitter splitter(sampleSize, kParameter);
+    for (UnsignedInteger foldIndex = 0; foldIndex < kParameter; ++foldIndex)
+    {
+      Indices indicesTest;
+      const Indices indicesTrain(splitter.generate(indicesTest));
+      const UnsignedInteger foldSize = indicesTest.getSize();
+      SymmetricMatrix projectionKFoldMatrix(foldSize);
+      for (UnsignedInteger i1 = 0; i1 < foldSize; ++i1)
+        for (UnsignedInteger i2 = 0; i2 < 1 + i1; ++i2)
+          projectionKFoldMatrix(i1, i2) = projectionMatrix(indicesTest[i1], indicesTest[i2]);
```

Comment:
This code could be improved in terms of performance if we have a new method `LeastSquaresImplementation.getHBlock(indices)` where `indices` is a list of rows. If a specific implementation is provide for Cholesky ([see}(https://github.com/openturns/openturns/blob/1d861ea09d429a01ae3d946ad99c2baac11b1495/lib/src/Base/Algo/CholeskyMethod.cxx#L304)), SVD ([see](https://github.com/openturns/openturns/blob/1d861ea09d429a01ae3d946ad99c2baac11b1495/lib/src/Base/Algo/SVDMethod.cxx#L210)), QR ([see](https://github.com/openturns/openturns/blob/1d861ea09d429a01ae3d946ad99c2baac11b1495/lib/src/Base/Algo/QRMethod.cxx#L149C11-L149C11)), this could improve the code.